packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -46,6 +46,11 @@ export async function execProgram(aws: SdkProvider, config: Configuration): Prom
   debug('context:', context);
   env[cxapi.CONTEXT_ENV] = JSON.stringify(context);
 
+  const build = config.settings.get(['build']);
+  if (build) {
+    await exec([build]);
```

Comment:
Seems like exec() treats the first item in the array as the command but this is not true in the case I described above. 

Can you add a test to verify ?

Reply:
You're right, the split doesn't happen in the case of the `build` key. It winds up working because the whole command just get put into `command` and `spawn()` happily accepts it.